pkg/logging/labels.go

```diff
@@ -1,15 +1,44 @@
 package logging
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/go-logr/logr"
 )
 
 // WithLabels decorates a logr.Logger so that any log entries contain all
 // labels which keys are prefix by labelKeyPrefix
 func WithLabels(logger logr.Logger, labels map[string]string, labelKeyPrefix string) logr.Logger {
 	for key, value := range labels {
-		logger = logger.WithValues(labelKeyPrefix+key, value)
+		logger = logger.WithValues(
+			fmt.Sprintf(
+				"%s_%s",
```

Comment:
Had a quick check and I think we might want to remove the underscore here, we already add this as part of the prefix.

```suggestion
				"%s%s",
```